Plutarch/Api/V1.hs

```diff
@@ -188,76 +190,143 @@ newtype PDatum (s :: S) = PDatum (Term s PData)
 
 instance PUnsafeLiftDecl PDatum where type PLifted PDatum = Plutus.Datum
 deriving via (DerivePConstantViaNewtype Plutus.Datum PDatum PData) instance (PConstant Plutus.Datum)
+deriving via (DerivePNewType PDatum PData) instance (PlutusType PDatum)
+deriving via (DerivePNewType PDatum PData) instance (PIsData PDatum)
 
 newtype PRedeemer (s :: S) = PRedeemer (Term s PData)
 
 instance PUnsafeLiftDecl PRedeemer where type PLifted PRedeemer = Plutus.Redeemer
 deriving via (DerivePConstantViaNewtype Plutus.Redeemer PRedeemer PData) instance (PConstant Plutus.Redeemer)
+deriving via (DerivePNewType PRedeemer PData) instance (PlutusType PRedeemer)
+deriving via (DerivePNewType PRedeemer PData) instance (PIsData PRedeemer)
 
 newtype PDatumHash (s :: S)
   = PDatumHash (Term s PByteString)
 
 instance PUnsafeLiftDecl PDatumHash where type PLifted PDatumHash = Plutus.DatumHash
 deriving via (DerivePConstantViaNewtype Plutus.DatumHash PDatumHash PByteString) instance (PConstant Plutus.DatumHash)
+deriving via (DerivePNewType PDatumHash PByteString) instance (PlutusType PDatumHash)
+deriving via (DerivePNewType PDatumHash PByteString) instance (PIsData PDatumHash)
 
 newtype PStakeValidatorHash (s :: S)
   = PStakeValidatorHash (Term s PByteString)
 
 instance PUnsafeLiftDecl PStakeValidatorHash where type PLifted PStakeValidatorHash = Plutus.StakeValidatorHash
 deriving via (DerivePConstantViaNewtype Plutus.StakeValidatorHash PStakeValidatorHash PByteString) instance (PConstant Plutus.StakeValidatorHash)
+deriving via (DerivePNewType PStakeValidatorHash PByteString) instance (PlutusType PStakeValidatorHash)
+deriving via (DerivePNewType PStakeValidatorHash PByteString) instance (PIsData PStakeValidatorHash)
 
 newtype PRedeemerHash (s :: S)
   = PRedeemerHash (Term s PByteString)
 
 instance PUnsafeLiftDecl PRedeemerHash where type PLifted PRedeemerHash = Plutus.RedeemerHash
 deriving via (DerivePConstantViaNewtype Plutus.RedeemerHash PRedeemerHash PByteString) instance (PConstant Plutus.RedeemerHash)
+deriving via (DerivePNewType PRedeemerHash PByteString) instance (PlutusType PRedeemerHash)
```

Comment:
I don't like the fact that we're using standalone derives here. If you flip around the parameters to `DerivePNewType`, it works fine without standalone derives.

Reply:
Oh yeah what the heck. Silly me, will fix this! I also don't like standalone deriving unnecessarily.

Reply:
Fixed!